### PR TITLE
Adding animation links for the *Transition widgets.

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds the animation links for illustrations of the Transition widgets, as well as adding a cross-references to the other transition widgets in the "See also" for each of their docs.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 1]
```diff
@@ -107,6 +107,27 @@ class _AnimatedState extends State<AnimatedWidget> {
...
+///  * [FadeTransition], a widget that animates the opacity of a widget.
+///  * [PositionedTransition], a widget that animates its child from a start
+///    position to an end position over the lifetime of the animation.
+///  * [RelativePositionedTransition], a widget that transitions its child's
+///    position based on the value of a rectangle relative to a bounding box.
+///  * [RotationTransition], a widget that animates the rotation of a widget.
+///  * [ScaleTransition], a widget that animates the scale of a transformed
+///    widget.
+///  * [SizeTransition], a widget that animates its own size and clips and
+///    aligns its child.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: i think we should be a bit more conservative in our cross-links here. For example, I don't see how SlideTransition really relates to DecoratedBoxTransition or FadeTransition (the others seem more related).

**Comment by Hixie** [comment]
LGTM modulo reducing the number of cross-references so as to focus on close relations. Ideally we would have a web where you can jump from one to the other to the other and they'd all be connected, without us having a fully-connected network which overwhelms the reader. :-)

**Comment by xster** [comment]
+1. A full interconnect seem a bit painful to maintain as well if adding a new transition type implies I need to go add it to every other transitions' docs. Adding to related transitions is less costly because hopefully the author already thoroughly went through the existing related ones to come to the conclusion that a new one needs to be added.

**Comment by gspencergoog** [comment]
OK, I'll reduce the interconnect.  It would be nice to have a way to do "meta" pages, so I could have a single "see also" to an intermediate page that linked all Transitions, and explained the entire class of widgets with a description.

**Comment by gspencergoog** [comment]
OK, updated.  PTAL.

**Comment by gspencergoog** [comment]
Oh, I missed the LGTM up there.  Landing on green.

**Pull request merged**